mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+                        np.expand_dims(losses_batch, axis=2),
+                        self.n_lambdas,
+                        axis=2
+                    )
+
+                    R = np.arange(self.n_lambdas) / self.n_lambdas
+                    K_R = np.cumsum(
+                        np.log(
+                            (
+                                1 -
+                                nu_batch *
+                                (losses_batch - R)
+                            ) +
+                            np.finfo(np.float64).eps
+                        ),
+                        axis=0
+                    )
+                    K_R = np.max(K_R, axis=0)
+                    K_R_max += K_R
+
+                r_hat_plus_tronc = (np.argwhere(
+                    np.cumsum(K_R_max > -np.log(delta), axis=1) == 1
+                )[:, 1] / self.n_lambdas)
+                r_hat_plus = np.ones(self.n_lambdas)
+                r_hat_plus[:len(r_hat_plus_tronc)] = r_hat_plus_tronc
+
+        else:
+            r_hat_plus = (n_obs / (n_obs + 1)) * r_hat + (1 / (n_obs + 1))
+
+        return r_hat, r_hat_plus
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieMultiLabelClassifier:
+        """
+        Fit the base estimator or use the fitted base estimator on
+        batch data. All the computed risks will be concatenated each
+        time the partial_fit method is called.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : NDArray of shape (n_samples, n_classes)
+            Training labels.
+
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no prediction sets.
+
+            By default None.
+
+        Returns
+        -------
+        MapieMultiLabelClassifier
+            The model itself.
+        """
+        # Checks
+        first_call = self._check_partial_fit_first_call()
+        self._check_parameters()
+        estimator = self._check_estimator(X, y, self.estimator)
+
+        X, y = indexable(X, y)
+        _check_y(y, multi_output=True)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
+        X = cast(NDArray, X)
+
+        self._check_all_labelled(y)
+        self.n_samples_ = _num_samples(X)
+
+        # Work
+        self.single_estimator_ = estimator
+        y_pred_proba = self.single_estimator_.predict_proba(X)
+        y_pred_proba_array = self._transform_pred_proba(y_pred_proba)
+        if first_call:
+            self.theta_ = np.zeros(X.shape[1])
```

Comment:
you are only interested in the shape of theta, no? why don't you store X.shape[1] ?

Reply:
I store X.shape[1]  into theta

Reply:
I followed the sklearn template

Reply:
You don't store only X.shape[1]. Why do yo use np.zeros () ?

Reply:
done